app/components/custom_status/custom_status_expiry.tsx

```diff
@@ -72,11 +72,11 @@ const CustomStatusExpiry = ({currentUser, isMilitaryTime, showPrefix, showTimeCo
             />
         );
     } else if (expiryMomentTime.isAfter(tomorrowEndTime)) {
-        let format = 'dddd';
+        let format: FormattedDateFormat = {weekday: 'long'};
         if (expiryMomentTime.isAfter(plusSixDaysEndTime) && isCurrentYear) {
-            format = 'MMM DD';
+            format = {month: 'short', day: 'numeric'};
         } else if (!isCurrentYear) {
-            format = 'MMM DD, YYYY';
+            format = {dateStyle: 'medium'};
```

Comment:
Since these are constant, can we move them outside of the component? The way I imagine it is something like...

```
const dateFormats = {
   withinWeek: {weekday: 'long'},
   withinYear: {month: 'short', day: 'numeric'},
   afterYear: {dateStyle: 'medium'};
}

const CustomStatusExpiry = (...) => {
    ...
        let format = dateFormats.withinWeek;
        ....
```

This way, we are passing always a constant value to `FromattedDate` instead of a new object on every render. Maybe FormattedDate is smart on the inside, or not too complex to re-render, but this helps being future proof.

Reply:
Sure !